packages/react/src/test/create-component_test.tsx

```diff
@@ -323,286 +305,159 @@ suite('createComponent', () => {
-    assert.equal(
-      el.getAttribute('aria-checked'),
-      wrappedEl.getAttribute('aria-checked')
-    );
+      render(<Tag hidden={undefined} />);
+      assert.equal(el.getAttribute('hidden'), null);
+      assert.equal(el.hidden, false);
 
-    await renderReactComponent({'aria-checked': true});
-    assert.equal(el.getAttribute('aria-checked'), 'true');
-    assert.equal(
-      el.getAttribute('aria-checked'),
-      wrappedEl.getAttribute('aria-checked')
-    );
+      render(<Tag hidden />);
+      assert.equal(el.getAttribute('hidden'), '');
+      assert.equal(el.hidden, true);
+    });
 
-    await renderReactComponent({'aria-checked': undefined});
-    assert.equal(el.getAttribute('aria-checked'), null);
-    assert.equal(
-      el.getAttribute('aria-checked'),
-      wrappedEl.getAttribute('aria-checked')
-    );
+    test(`${name}: set/unset enumerated attributes`, async () => {
+      render(<Tag />);
+      el = container.querySelector(name)!;
+      assert.equal(el.getAttribute('draggable'), null);
+      assert.equal(el.draggable, false);
+
+      render(<Tag draggable />);
+      assert.equal(el.getAttribute('draggable'), 'true');
+      assert.equal(el.draggable, true);
+
+      render(<Tag draggable={false} />);
+      assert.equal(el.getAttribute('draggable'), 'false');
+      assert.equal(el.draggable, false);
+
+      render(<Tag draggable />);
+      assert.equal(el.getAttribute('draggable'), 'true');
+      assert.equal(el.draggable, true);
+
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      render(<Tag draggable={null as any} />);
+      assert.equal(el.getAttribute('draggable'), null);
+      assert.equal(el.draggable, false);
+
+      render(<Tag draggable />);
+      assert.equal(el.getAttribute('draggable'), 'true');
+      assert.equal(el.draggable, true);
+
+      render(<Tag draggable={undefined} />);
+      assert.equal(el.getAttribute('draggable'), null);
+      assert.equal(el.draggable, false);
+
+      render(<Tag draggable="true" />);
+      assert.equal(el.getAttribute('draggable'), 'true');
+      assert.equal(el.draggable, true);
+
+      // The following test fails for basic-element only
+      // See https://github.com/lit/lit/issues/4391
+      // render(<Tag draggable="false" />);
+      // assert.equal(el.getAttribute('draggable'), 'false');
+      // assert.equal(el.draggable, false);
+    });
 
-    await renderReactComponent({'aria-checked': true});
-    assert.equal(el.getAttribute('aria-checked'), 'true');
-    assert.equal(
-      el.getAttribute('aria-checked'),
-      wrappedEl.getAttribute('aria-checked')
-    );
-  });
+    test(`${name}: sets/unsets boolean aria attributes`, async () => {
+      render(<Tag />);
+      el = container.querySelector(name)!;
+      assert.equal(el.getAttribute('aria-checked'), null);
 
-  test('unsets reflected attributes from property with different specifier on null or undefined prop', async () => {
```

Comment:
No I think it was a bad test to begin with. It's bad prop usage that TS needed to be bypassed to set anyway. React users should be setting `aria-` attributes, rather than properties.